src/chain.rs

```diff
@@ -67,7 +75,15 @@ pub trait PevmChain: Debug {
         txs: &BlockTransactions<Self::Transaction>,
         tx_results: &[PevmTxExecutionResult],
     ) -> B256;
+
+    /// Build Self::Transaction type from Alloy's transaction
+    fn cast_from_eth_transaction(&self, tx: alloy_rpc_types::Transaction) -> Self::Transaction;
```

Comment:
`eth_transaction` is wrong. For instance, Ethereum transactions don't have the `signature` that `alloy_rpc_types::Transaction` supports. The input here can keep growing to cover more chains, and that's why it can be converted to an arbitrary `Self::Transaction` here.

The more precise name would be `build_tx_from_{alloy|rpc}_tx`.